lib/logger.js

```diff
@@ -8,15 +8,12 @@ function Logger() {
 	Object.keys(Logger.prototype).forEach(function (key) {
 		this[key] = this[key].bind(this);
 	}, this);
+
+	this.reporter = reporter;
```

Comment:
Is there any particular reason the `Logger` needs to know about `api`? I'm not familiar with the overall design, but it seems to me that `Logger` only contains `api` in order to pass it to the different reporters, and does not use it on its own. It's a bit confusing to me.